stream-compositions/api/cursors-api/transaction-cursor-api/src/main/resources/service-api-v2.yaml

```diff
@@ -0,0 +1,267 @@
+      operationId: getById
+      parameters:
+        - name: id
+          in: path
+          required: true
+          description: internal id of the cursor
+          schema:
+            type: string
+
+      responses:
+        "200":
+          description: Transaction Cursor Details.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TransactionCursorResponse'
+        "400":
+          description: BadRequest
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BadRequest'
+        "500":
+          description: InternalServerError
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InternalServerError'
+  /service-api/v2/cursor/arrangement/{arrangementId}:
+    get:
+      tags:
+        - TransactionCursor
+      summary: Gets Cursor by the internal Arrangement id.
+      operationId: getByArrangementId
+      parameters:
+        - name: arrangementId
+          in: path
+          required: true
+          description: internal id of the arrangement
+          schema:
+            type: string
+      responses:
+        "200":
+          description: Transaction Cursor Details.
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TransactionCursorResponse'
+        "400":
+          description: BadRequest
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BadRequest'
+        "500":
+          description: InternalServerError
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InternalServerError'
+    patch:
+      tags:
+        - TransactionCursor
+      summary: Patches Transaction Cursor by arrangement id.
+      description: Patches Transaction Cursor by Arrangement Id.
+      operationId: patchByArrangementId
+      parameters:
+        - name: arrangementId
+          in: path
+          required: true
+          description: internal id of the arrangement
+          schema:
+            type: string
+      requestBody:
+        description: Patch Transaction Cursor
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/TransactionCursorPatchRequest'
+      responses:
+        "200":
+          description: Transaction Cursor Patched.
+        "400":
+          description: BadRequest
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/BadRequest'
+        "500":
+          description: InternalServerError
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/InternalServerError'
+  /service-api/v2/cursor/upsert:
+    post:
```

Comment:
why aren't we using a put for upserting?

Reply:
Put is for updates, but upsert would update/insert so using post here. All upsert/refresh operations are POST
DBS API follow the same pattern.

Reply:
https://developer.backbase.com/api/specs/arrangement-manager/arrangement-service-api/2.6.0/operations/Arrangements/postBatchUpsertArrangements